didcomm/did_doc/did_doc.py

```diff
@@ -112,7 +119,7 @@ def verification_method(self, kid: DID_URL) -> Optional[VerificationMethod]:
         pass
 
     @abstractmethod
-    def services(self) -> List[DIDDocService]:
+    def services(self) -> List[DIDCommService]:
```

Comment:
I still think that generic DID API is  better than DID Comm specific DID API.

Reply:
1) DIDComm lib doesn't care and doesn't process any other types of services besides DIDCommService
2) The interface is a subset of DID DOC needed for DIDComm needs. So, the user doesn't need to process the parts of DID DOC that are not needed for DID Comm.
3) Having a generic Service class will make the API in Python more complicated. How sholuld we support DIDCommService type? Have just one interface for all Services and make routing key an optional field that will be None for non-DIDCOmmServices? It will look weird. Use inheritance and have a dedicated class for DIDCommService extending  base Service class? There is a chance then that all user will use Service class instead of DIDCommService and we will not have routing keys value at all.